index.bs

```diff
@@ -787,87 +782,92 @@ Issue: Should fake reports respect the user agent's [=max reports per attributio
-1. Let |eventConfig| be a new [=event-level trigger configuration=] with the items:
-    : [=event-level trigger configuration/trigger data=]
-    :: |eventSourceTriggerData|
-    : [=event-level trigger configuration/dedup key=]
-    :: |dedupKey|
-    : [=event-level trigger configuration/priority=]
-    :: |triggerPriority|
-    : [=event-level trigger configuration/filters=]
-    :: «[ "`source_type`" → « "`event`" » ]»
+To <dfn>parse event triggers</dfn> given an [=ordered map=] |map|:
+
+1. Let |eventTriggers| be a new [=set=].
+1. If |map|["`event_trigger_data`"] does not [=map/exists|exist=], return
+    |eventTriggers|.
+1. Let |values| be |map|["`event_trigger_data`"].
+1. If |values| is not a [=list=], return null.
+1. [=list/iterate|For each=] |value| of |values|:
+    1. If |value| is not an [=ordered map=], return null.
+    1. If |value|["`trigger_data`"] does not [=map/exists|exist=] or is not a
+        [=string=], return null.
+    1. Let |triggerData| be the result of applying the
+        <a spec="html">rules for parsing non-negative integers</a> to
+        |value|["`trigger_data`"].
+    1. If |triggerData| is an error, set |triggerData| to 0.
+    1. Let |dedupKey| be null.
+    1. If |value|["`deduplication_key`"] [=map/exists=] and is a [=string=]:
+        1. Set |dedupKey| to the result of applying the
+            <a spec="html">rules for parsing non-negative integers</a> to
+            |value|["`deduplication_key`"].
+        1. If |dedupKey| is an error, set |dedupKey| to null.
+    1. Let |priority| be 0.
+    1. If |value|["`priority`"] [=map/exists=] and is a [=string=]:
+        1. Set |priority| to the result of applying the
+            <a spec="html">rules for parsing integers</a> to
+            |value|["`priority`"].
+        1. If |priority| is an error, set |priority| to 0.
+    1. Let |eventTrigger| be a new [=event-level trigger configuration=] with
+        the items:
+        : [=event-level trigger configuration/trigger data=]
+        :: |triggerData|
+        : [=event-level trigger configuration/dedup key=]
+        :: |dedupKey|
+        : [=event-level trigger configuration/priority=]
+        :: |priority|
+        : [=event-level trigger configuration/filters=]
+        :: «[]»
+        : [=event-level trigger configuration/negated filters=]
+        :: «[]»
+    1. [=set/Append=] |eventTrigger| to |eventTriggers|.
+1. Return |eventTriggers|.
+
+Issue: Parse filters and negated filters.
+
+To <dfn noexport>parse trigger-registration JSON</dfn> given a [=string=]
+|json|, a [=site=] |destination|, and an [=origin=] |reportingOrigin|:
+
+1. Assert: |reportingOrigin| is a [=potentially trustworthy origin=].
+1. Let |value| be the result of running
+    [=parse a JSON string to an Infra value=] with |json|.
+1. If |value| is not an [=ordered map=], return null.
+1. Let |eventTriggers| be the result of running [=parse event triggers=]
+    with |value|.
+1. If |eventTriggers| is null, return null.
```

Comment:
This is assuming https://github.com/WICG/attribution-reporting-api/pull/448 is landed, but will also need to change when we add aggregate parsing, correct?

Reply:
Yes.